src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -72,44 +73,75 @@ public ProductService(BaseDTO dto) {
         return components;
     }
 
+    private Set<String> getSupportedLocales() {
+        return getSupportedLocales(true);
+    }
+
+    public Set<String> getCachedSupportedLocales() {
```

Comment:
I also agree cache locale list when initialization no matter initializeCache is true or false, since it may reduce the chance of locale list fetching be triggered by message fetching which can improve performance at message fetching.
Further more, we may remove fetch locale list task code at message fetching.

This can be done in a new PR.

Reply:
1. We should not force to cache this at app start up. This is why we have initializeCache property in the config. Even component messages are sometimes not initialized.
2. In some cases, this will always fail (FileSystems.newFileSystem walk fails in nested jar applications like in Atlas for Java 11 or less), so always calling getSupportedLocales that fails might affect performance a it tries to read the disk.

Reply:
Please check ProductService. The code is now more unified.